### PR TITLE
chore(weave): Implement clickhouse client

### PR DESCRIPTION
**Pull request by tssweeney**
Pairs with https://github.com/wandb/core/pull/19541

Task list moved to: https://www.notion.so/wandbai/Weaveflow-Clickhouse-Migration-Project-a65d768d153a4779b4b34a1b22060cf4

### CODE REVIEW

File: weave/monitoring/openai/openai.py [hunk 1]
```diff
@@ -114,10 +114,9 @@ def patch() -> None:
     def _patch() -> None:
         unpatch_fqn = f"{unpatch.__module__}.{unpatch.__qualname__}()"
 
-        if _get_global_monitor() is not None:
-            # info(f"Patching OpenAI completions.  To unpatch, call {unpatch_fqn}")
-
-            gc = graph_client_context.require_graph_client()
+        gc = graph_client_context.require_graph_client()
+        if gc:
+            info(f"Patching OpenAI completions.  To unpatch, call {unpatch_fqn}")
```
Comment (shawnlewis) [inline weave/monitoring/openai/openai.py:119]: We don't want this to be printed.

File: weave/op_def_type.py [hunk 2]
```diff
@@ -187,12 +187,18 @@ class RefJSONEncoder(json.JSONEncoder):
     SPECIAL_REF_TOKEN = "__WEAVE_REF__"
 
     def default(self, o):
+        from .trace_server.graph_client_trace import TraceNounRef
```
Comment (shawnlewis) [inline weave/op_def_type.py:190]: First time encountering "TraceNounRef" for me. It feels like an odd name (unexpected)

File: weave/trace_server/graph_client_trace.py [hunk 3]
```diff
@@ -0,0 +1,707 @@
...
+        return [refs_to_str(v) for v in val]  # type: ignore
+    else:
+        return val
+
+
+# `wandb-trace:///[entity]/[project]/call/[ID]`
+# `wandb-trace:///[entity]/[project]/op/[name]:[CONTENT_HASH]`
+# `wandb-trace:///[entity]/[project]/obj/[name]:[CONTENT_HASH]/[PATH]#[EXTRA]`
+@dataclasses.dataclass
+class TraceNounUri(uris.WeaveURI):
```
Comment (shawnlewis) [inline weave/trace_server/graph_client_trace.py:165]: Seems like we could just say "TraceURI"

**Review by shawnlewis — APPROVED** [review]

File: weave/monitoring/openai/openai.py (hunk 1 above)
Comment (tssweeney) [reply weave/monitoring/openai/openai.py:119]: fixing

File: weave/trace_server/graph_client_trace.py (hunk 3 above)
Comment (tssweeney) [reply weave/trace_server/graph_client_trace.py]: fixed

File: weave/op_def_type.py (hunk 2 above)
Comment (tssweeney) [reply weave/op_def_type.py:190]: fixed

**Pull request merged**